vllm/core/scheduler.py

```diff
@@ -83,18 +83,19 @@ def __init__(
         # LoRAs. This should be improved in the future.
         self.lora_config = lora_config
 
-        self.prompt_limit = min(self.scheduler_config.max_model_len,
-                                self.scheduler_config.max_num_batched_tokens)
+        if self.scheduler_config.embedding_mode:
+            # Profile this value if necessary
+            self.prompt_limit = self.scheduler_config.max_num_batched_tokens
+        else:
+            self.prompt_limit = min(
+                self.scheduler_config.max_model_len,
+                self.scheduler_config.max_num_batched_tokens)
```

Comment:
is it possible to use the same abstraction for both model types? trying to reduce branches in the codebase

Reply:
Removed this.